File: Csharp/chapter01/knock02.cs
/*
 * Chapter01, knock02
 * Task: concat two strings alternately
 */

using System;


public class ConcatString {
    static public void Main() {
        string word1 = "パトカー";
        string word2 = "タクシー";
        Console.WriteLine(Concat(word1, word2));
    }


    public static string Concat(string str1, string str2) {
        string concatString = "";
        for (int i=0; i < System.Math.Max(str1.Length, str2.Length); i ++) {
            concatString += str1[i];
            concatString += str2[i];
        }
        return concatString;
    }
}

File: Ruby/Chapter01/knock02.rb
# str1 = "パトカー", str2 = "タクシー"
# alternately connect str1 and str2 : = "パタトクカシーー"


str1, str2 = "パトカー", "タクシー"
connected = ""

0.upto(str1.length - 1) {|num|
  connected += str1[num] + str2[num]
}
puts(connected)
